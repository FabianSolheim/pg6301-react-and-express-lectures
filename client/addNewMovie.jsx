import React, { useState } from "react";

export function AddNewMovie() {
  const [title, setTitle] = useState("");
  return (
    <form>
      <h1>Add new movie</h1>

      <div>
        <label>
          <strong>Title:</strong>{" "}
          <input value={title} onChange={(e) => setTitle(e.target.value)} />
        </label>
      </div>
    </form>
  );
}
