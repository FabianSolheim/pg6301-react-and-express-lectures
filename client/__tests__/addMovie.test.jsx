import { AddNewMovie } from "../addNewMovie";
import React from "react";
import ReactDOM from "react-dom";

describe("add movie component", () => {
  it("shows movies form", () => {
    const element = document.createElement("div");
    ReactDOM.render(<AddNewMovie />, element);
    expect(element.innerHTML).toMatchSnapshot();
    expect(
      Array.from(element.querySelectorAll("form label strong")).map(
        (e) => e.innerHTML
      )
    ).toEqual(["Title:", "Year:", "Country:", "Plot:"]);
  });
});
