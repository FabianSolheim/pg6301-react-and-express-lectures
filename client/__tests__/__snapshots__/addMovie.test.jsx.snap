// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add movie component shows movies form 1`] = `"<form><h1>Add new movie</h1></form>"`;
